### PR TITLE
added unix-sockets feature support

### PR DESCRIPTION
**Pull request by gngeorgiev**
Related to https://github.com/alexcrichton/curl-rust/issues/297

Example:
```
let mut easy = Easy::new();
easy.unix_socket("/var/socket_path.sock")?;
easy.url("http://localhost/path")?;
```

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR!

I think though it's probably fine to enable this unconditionally, could this just be set always on unix platforms in the build script?

**Comment by gngeorgiev** [comment]
@alexcrichton I think this is reasonable. I will do the changes!

**Comment by SecurityInsanity** [comment]
Hey 👋 

Looks like it been a bit since this PR was touched (I'm assuming due to holidays). However I'll ask just in case, anything I can do to help get this feature in, and released?

**Comment by gngeorgiev** [comment]
@SecurityInsanity Sorry, yes holidays + new job are the result of my forgetfulness about this. Will do my best to finish it during the weekend ✌️

**Comment by SecurityInsanity** [comment]
Hey any update on this? Perhaps I can take it over? I'd like to get off my forks :sweat_smile:

**Comment by gngeorgiev** [comment]
@alexcrichton, @SecurityInsanity could you take a look now? I hope this is closer to what @alexcrichton wanted 🙇

**Comment by gngeorgiev** [comment]
I spent some time debugging the `mingw` failure. Right now I don't know how to fix the missing `un.h` file to `mingw`. It seems that other headers are there, e.g. `types.h` but `un.h` isn't. 

```shell
root@d9242764c267:/# find / -name un.h
/usr/include/linux/un.h
/usr/include/x86_64-linux-gnu/sys/un.h
root@d9242764c267:/# find / -name types.h
/usr/include/linux/types.h
/usr/include/linux/iio/types.h
/usr/include/x86_64-linux-gnu/asm/types.h
/usr/include/x86_64-linux-gnu/bits/types.h
/usr/include/x86_64-linux-gnu/sys/types.h
/usr/include/rpc/types.h
/usr/include/asm-generic/types.h
/usr/lib/gcc/x86_64-w64-mingw32/5.3-win32/include/c++/parallel/types.h
/usr/lib/gcc/x86_64-w64-mingw32/5.3-posix/include/c++/parallel/types.h
/usr/lib/gcc/i686-w64-mingw32/5.3-win32/include/c++/parallel/types.h
/usr/lib/gcc/i686-w64-mingw32/5.3-posix/include/c++/parallel/types.h
/usr/share/mingw-w64/include/sys/types.h
/usr/i686-w64-mingw32/include/sys/types.h
/usr/x86_64-w64-mingw32/include/sys/types.h
```

@alexcrichton Do you have any idea about this 🤔

File: curl-sys/build.rs [hunk 1]
```diff
@@ -212,6 +212,11 @@ fn main() {
             .file("curl/lib/vauth/vauth.c");
     }
 
+    if cfg!(not(windows)) {
```
Comment (alexcrichton) [inline curl-sys/build.rs:215]: I think you'll need to check env vars instead of `cfg!` here because of cross-compilation.

**Comment by gngeorgiev** [comment]
Sorry, @alexcrichton I have no idea why [x86_64 nighly is failing](https://github.com/alexcrichton/curl-rust/pull/309/checks?check_run_id=575328522#step:6:2281). Maybe it's a nightly thing since beta is passing? I have rebased upstream on top of my work btw 🙂

**Comment by alexcrichton** [comment]
Yes I've tried to [fix that on master now](https://github.com/alexcrichton/curl-rust/commit/68349087c8c09a5e56b57bcfc40c12e790aa5c71), mind rebasing to pull that in?

**Comment by SecurityInsanity** [comment]
I realize things are a bit focused elsewhere right now, so don't feel like this needs to be done immediately or anything, but figured I'd bump again for visibility. If I can be of any help getting this across the line, I'd also be happy to help.

**Comment by SecurityInsanity** [comment]
@gngeorgiev , no rush but any update? Would it be easier if I opened up my own pr and we can close this one, if you have other things going on? Don't wanna step on any toes, and realize there are many things going on right now, but also really tired of maintaining a manual fork of libraries :sweat_smile:

**Comment by sagebind** [comment]
@SecurityInsanity I'm OK with you taking this over if @gngeorgiev doesn't have time to work on this right now. Feel free to open an alternative PR when you have the opportunity.

**Comment by SecurityInsanity** [comment]
https://github.com/alexcrichton/curl-rust/pull/346 has been opened, and has been updated to latest master, and I can keep it updated. I've cherrypicked though to keep @gngeorgiev 's credit for the work they did do.

**Comment by sagebind** [comment]
#346 has been merged and is now available in curl 0.4.31. Thanks to @gngeorgiev and @SecurityInsanity!

**Pull request closed without merging**